### PR TITLE
feat: Add `vultr-cli` spec

### PR DESCRIPTION
**Pull request by rmorey**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
Adds spec for the [Vultr CLI](https://github.com/vultr/vultr-cli)

Vultr is a cloud services provider (VPS, object storage, etc) and this is their offical CLI.

**What is the current behavior? (You can also link to an open issue here)**

**What is the new behavior (if this is a feature change)?**

**Additional info:**

### CODE REVIEW

**Comment by QuiiBz** [comment]
Hello @rmorey, thanks for opening this PR! Is it still in draft, or ready for review?

**Comment by rmorey** [comment]
> Hello @rmorey, thanks for opening this PR! Is it still in draft, or ready for review?

Still in draft! :)

**Comment by QuiiBz** [comment]
> Still in draft! :)

Alright, feel free to @ me when you're ready for a review!

File: dev/vultr-cli.ts [hunk 1]
```diff
@@ -0,0 +1,562 @@
...
+  },
+};
+
+const createOptions = [
+  {
+    name: ["--os", "-o"],
+    description:
+      "ID of the operating system that will be installed on the server",
+    isRequired: true,
+    args: {},
```
Comment (fedeci) [inline dev/vultr-cli.ts:31]: Could you provide a name for the `args` please? Both here and in the other parts of the spec😄

File: dev/vultr-cli.ts [hunk 2]
```diff
@@ -0,0 +1,562 @@
...
+        .split("\n")
+        .slice(1, -3)
+        .map((line) => ({
+          name: line.split("\t")[0],
+          description: line.split("\t").slice(1, -4).join("\t"),
+        })),
+  },
+};
+
+const createOptions = [
```
Comment (fedeci) [inline dev/vultr-cli.ts:25]:
```suggestion
const createOptions: Fig.Option[] = [
```

File: dev/vultr-cli.ts [hunk 3]
```diff
@@ -0,0 +1,562 @@
...
+        },
+        { name: "start", description: "Start a bare metal server" },
+        {
+          name: "user-data",
+          description:
+            "User-data is used to access bare metal server user-data commands",
+        },
+        {
+          name: "vnc",
+          description: "Get a bare metal server's VNC url by <bareMetalID>",
```
Comment (fedeci) [inline dev/vultr-cli.ts:183]: Does it accept a `bareMetalId` arg?
```suggestion
          description: "Get a bare metal server's VNC url by <bareMetalID>",
          args: bareMetalIdArg,
```

**Review by fedeci — COMMENTED** [review]
Thank you!

File: dev/vultr-cli.ts [hunk 4]
```diff
@@ -0,0 +1,563 @@
+const bareMetalIdArg: Fig.Arg = {
+  name: "bareMetalId",
+  description: "Bare Metal ID",
+  // TODO: suggestions
```
Comment (QuiiBz) [inline dev/vultr-cli.ts:4]: Are you going to implement the suggestions in this PR?

**Review by QuiiBz — CHANGES_REQUESTED** [review]

**Pull request merged**